### PR TITLE
Work winxin login and department

### PR DESCRIPTION
**Pull request by sniper-py**

### CODE REVIEW

File: seahub/work_weixin/views.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from seahub.utils import get_service_url
+from seahub.auth.models import SocialAuthUser
+
+logger = logging.getLogger(__name__)
+
+
+# # uid = corpid + '_' + userid
+
+
+def work_weixin_oauth_login(request):
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: 1,  not allowed creeate unknow user:    Error, new user registration is not allowed, please contact administrator.
1,  not auto active: Your account is created successfully, please wait for administrator to activate your account.
1,  not enable oauth login: Feature disabled.
1,  enable oauth login but wrong congure: Error, please contact administrator.

File: seahub/work_weixin/views.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+def work_weixin_oauth_callback(request):
+    if not work_weixin_oauth_check():
+        return render_error(request, _('Error, please contact administrator.'))
+
+    code = request.GET.get('code', None)
+    state = request.GET.get('state', None)
+    if state != request.session.get('oauth_state', None) or not code:
+        return render_error(request, '取消授权')
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: return render_error(request, _('Error, please contact administrator.'))

File: seahub/work_weixin/views.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        return render_error(request, _('Error, please contact administrator.'))
+
+    data = {
+        'access_token': access_token,
+        'code': code,
+    }
+    api_response = requests.get(WORK_WEIXIN_GET_USER_INFO_URL, params=data)
+    api_response_dic = handler_work_weixin_api_response(api_response)
+    if not api_response_dic:
+        return render_error(request, _('Error, please contact administrator.'))
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: if need log

File: seahub/work_weixin/views.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        email = gen_user_virtual_id()
+        work_weixin_user = SocialAuthUser.objects.add_by_uid(email, WORK_WEIXIN_PROVIDER, uid)
+
+    try:
+        user = auth.authenticate(remote_user=email)
+    except User.DoesNotExist:
+        user = None
+
+    if not user:
+        return render_error(request, _('创建账户失败，请联系管理员'))
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: return render_error(request, _('Error, please contact administrator.'))

File: seahub/work_weixin/views.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        return render_error(request, _('Error, please contact administrator.'))
+
+    if not api_response_dic.get('UserId', None):
+        logger.error(json.dumps(api_response_dic))
+        return render_error(request, _('Error, please contact administrator.'))
+
+    user_id = api_response_dic.get('UserId')
+    uid = WORK_WEIXIN_UID_PREFIX + user_id
+
+    work_weixin_user = SocialAuthUser.objects.get_by_uid(WORK_WEIXIN_PROVIDER, uid)
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: try:
return render_error(request, _('Error, please contact administrator.'))

File: seahub/work_weixin/views.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    if not user.is_active:
+        return render_error(request, _('账户未激活，请联系管理员'))
+
+    # update user info
+    user_info_data = {
+        'access_token': access_token,
+        'userid': user_id,
+    }
+    user_info_api_response = requests.get(WORK_WEIXIN_GET_USER_PROFILE_URL, params=user_info_data)
```
Comment (imwhatiam) [inline seahub/work_weixin/views.py]: update user info setting todo.

File: frontend/src/pages/sys-admin/work-weixin-departments.js [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+        });
+      } else {
+        this.setState({
+          newUsersTempObj: {},
+        });
+      }
+    });
+  };
+
+  handlerSubmit = () => {
```
Comment (freeplant) [inline frontend/src/pages/sys-admin/work-weixin-departments.js]: handlerSubmit -> handleSubmit

**Pull request merged**